pkg/sypgp/sypgp.go

```diff
@@ -351,6 +351,76 @@ func StorePubKey(e *openpgp.Entity) (err error) {
 	return
 }
 
+// compareLocalPubKey compares a key ID with a string, returning true if the
+// key and oldToken match.
+func compareLocalPubKey(e *openpgp.Entity, oldToken string) bool {
+	return fmt.Sprintf("%X", e.PrimaryKey.Fingerprint) == oldToken
+}
+
+// CheckLocalPubKey will check if we have a local public key matching ckey string
+// returns true if there's a match.
+func CheckLocalPubKey(ckey string) (bool, error) {
+	f, err := os.OpenFile(PublicPath(), os.O_CREATE|os.O_RDONLY, 0600)
+	if err != nil {
+		return false, fmt.Errorf("unable to open local keyring: %v", err)
+	}
+	defer f.Close()
+
+	// read all the local public keys
+	elist, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		return false, fmt.Errorf("unable to list local keyring: %v", err)
+	}
+
+	for i := range elist {
+		return compareLocalPubKey(elist[i], ckey), nil
```

Comment:
Could there be more than one key in the keyring? It looks like this loop only checks the first one and returns.

Reply:
Cuckoo -> @WestleyK